base/services/telnetd/telnetd.c

```diff
@@ -367,12 +367,14 @@ static void RunShell(client_t *client)
    STARTUPINFO           si;
    PROCESS_INFORMATION   piProcInfo;
    SECURITY_ATTRIBUTES   saAttr;
+   char cmd_path[MAX_PATH];
+
+   if (!GetEnvironmentVariableA("COMSPEC", cmd_path, ARRAYSIZE(cmd_path)))
+   {
+      GetSystemDirectoryA(cmd_path, ARRAYSIZE(cmd_path));
+      strcat(cmd_path, "\\cmd.exe");
```

Comment:
Sorry, forgot to review the telnet part... Can you also use StringCchCatA here and check GetSystemDirectory return?

Reply:
OK.